rdmo/core/settings.py

```diff
@@ -50,7 +50,8 @@
     'django.contrib.auth.middleware.AuthenticationMiddleware',
     'django.contrib.messages.middleware.MessageMiddleware',
     'django.middleware.clickjacking.XFrameOptionsMiddleware',
-    'django.contrib.sites.middleware.CurrentSiteMiddleware'
+    'django.contrib.sites.middleware.CurrentSiteMiddleware',
+    'allauth.account.middleware.AccountMiddleware',
```

Comment:
allauth depencencies are not in the core of RDMO this needs to go in the (local) settings in `rdmo-app`, like the Shibboleth stuff.

Reply:
I guess people need to add `MIDDLEWARE = [*MIDDLEWARE, 'allauth.account.middleware.AccountMiddleware']` to their `config/settings/local.py`.

Reply:
Otherwise the middleware would run on every request, even when only Shibboleth or LDAP is used. Which is not a catastrophe but kind of ugly.

Reply:
Okay.
I just took all the requirements from the master branch and put those in the toml file. I said allauth is a main dependency, not optional:
https://github.com/rdmorganiser/rdmo/blob/44dcda2f419f6dff269ea5e0128cbcdf0d77006b/requirements.txt#L5

Also it is used in the testing config:
https://github.com/rdmorganiser/rdmo/blob/44dcda2f419f6dff269ea5e0128cbcdf0d77006b/testing/config/settings/base.py#L21

I added it the middleware, because otherwise the tests failed. The middleware needs to updated in testing config then.

Right?

I will do that later.

Reply:
With primary way of authentication I mean that most instances will use allauth.

Reply:
Alright, I see. That would be the solution?

- move django-allauth from dependencies to optional-dependencies
- remove the middleware
- add it to documentation of rdmo-app

Did I miss something?

Reply:
Yes, I would create a speperate group:

```
allauth = [
  "django-allauth~=0.53.1",
]
```

I just tried it with and without allauth, and without allauth only the allauth specific tests for password change and signup fail.

Reply:
I plan to update the docs on my remaining train rides this week.

Reply:
I can look at the docs in the week after ;)

Reply:
All done.